core/ledger/kvledger/txmgmt/privacyenabledstate/snapshot.go

```diff
@@ -65,30 +45,61 @@ func (s *DB) ExportPubStateAndPvtStateHashes(dir string, newHashFunc snapshot.Ne
 		}
 		switch {
 		case isHashedDataNs(compositeKey.Namespace):
+			if pvtStateHashesWriter == nil { // encountered first time the pvt state hash element
+				pvtStateHashesWriter, err = newSnapshotWriter(
+					filepath.Join(dir, pvtStateHashesFileName),
+					filepath.Join(dir, pvtStateHashesMetadataFileName),
+					dbValueFormat,
+					newHashFunc,
+				)
+				if err != nil {
+					return nil, err
+				}
+				defer pvtStateHashesWriter.close()
+			}
 			if err := pvtStateHashesWriter.addData(compositeKey, dbValue); err != nil {
 				return nil, err
 			}
 		default:
+			if pubStateWriter == nil { // encountered first time the pub state element
+				pubStateWriter, err = newSnapshotWriter(
```

Comment:
Here also, the public state would be non-empty even at block height 1 as we store the genesis block. Earlier code structure -- file creation, dumping data to file, writing metadata, and closing it looked neater. If no data is dumped, it was reflected in the metadata. 

If we still decide not to have empty files, having `IsEmpty()` at each store would make this more explicit and neat IMO.

Reply:
The same response as in the [comment](https://github.com/hyperledger/fabric/pull/1345/files#r431606398). Regarding `IsEmpty`, how do you handle for private state hashes?